page-objects/pages/HMWRKHomeworkListPage.js

```diff
@@ -85,24 +84,26 @@ module.exports = {
         return listOfTasks.length > 0 ? true : false
     },
 
-    getAllTasks: async function () {
+    getListOfTaskTitles: async function () {
         await waitHelpers.waitUntilElementIsNotVisible(".loaded #MathJax_Message");
-	    const selector = taskBox;
-	    try {
-		    await waitHelpers.waitUntilElementIsPresent(selector);
-	    } catch (err) {
-		    return [];
-	    }
-	    const listOfTaskElements = await driver.$$(selector);
-	    let taskTitleList = await Promise.all(listOfTaskElements.map(async (element) => await element.getText()));
-	    return taskTitleList;
+	    return elementHelpers.getTextFromAllElements(taskTitleContainer);
     },
 
 
     isTaskVisible: async function (taskname) {
-        const allTasks = await this.getAllTasks();
+        const allTasks = await this.getListOfTaskTitles();
         const isTaskOnList = allTasks.some((element) => element.includes(taskname));
-        return isTaskOnList;
+        const msg = 'Task with name is not visible on the list: \n';
+        const resultMsg = 'Expected: ' + taskname + ', Actual: ' + allTasks;
+        await expect(isTaskOnList, msg + resultMsg).to.equal(true);
+    },
+
+    isTaskNotVisible: async function (taskname) {
+        const allTasks = await this.getListOfTaskTitles();
+        const isTaskOnList = allTasks.some((element) => element.includes(taskname));
+        const msg = 'Task with name is not visible on the list: \n';
```

Comment:
Message should be different in this case, or?

Reply:
Yes I will fix it today